rustls/src/crypto/mod.rs

```diff
@@ -226,45 +222,39 @@ impl CryptoProvider {
     /// Call this early in your process to configure which provider is used for
     /// the provider.  The configuration should happen before any use of
     /// [`ClientConfig::builder()`] or [`ServerConfig::builder()`].
-    #[cfg(feature = "std")]
     pub fn install_default(self) -> Result<(), Arc<Self>> {
-        PROCESS_DEFAULT_PROVIDER.set(Arc::new(self))
+        static_default::install_default(Arc::new(self))
     }
 
-    /// Sets this `CryptoProvider` as the default for this process.
-    ///
-    /// This can be called successfully at most once in any process execution.
+    /// Returns the default `CryptoProvider` for this process.
     ///
-    /// Call this early in your process to configure which provider is used for
-    /// the provider.  The configuration should happen before any use of
-    /// [`ClientConfig::builder()`] or [`ServerConfig::builder()`].
-    #[cfg(not(feature = "std"))]
-    pub fn install_default(self) -> Result<(), Box<Arc<Self>>> {
-        PROCESS_DEFAULT_PROVIDER.set(Box::new(Arc::new(self)))
+    /// This will be `None` if no default has been set yet.
+    #[cfg(not(feature = "withrcalias"))]
+    pub fn get_default() -> Option<&'static Arc<Self>> {
+        static_default::get_default()
     }
 
     /// Returns the default `CryptoProvider` for this process.
     ///
     /// This will be `None` if no default has been set yet.
-    pub fn get_default() -> Option<&'static Arc<Self>> {
-        PROCESS_DEFAULT_PROVIDER.get()
+    #[cfg(feature = "withrcalias")]
+    pub fn get_default() -> Option<Arc<Self>> {
+        static_default::get_default()
     }
```

Comment:
Unfortunately this API function has a somewhat different return type depending on the build configuration.

Existing return type with static-lifetime reference to `Arc` cannot be replicated with `Rc` alias due to differences in storing & cloning.

I would rather not break existing API usage for any existing users.

I think I need to add a note to the source to explain this.